tests/src/test/java/alluxio/client/fuse/JNIFuseIntegrationTest.java

```diff
@@ -43,4 +52,169 @@ public void mountFuse(FileSystem fileSystem, String mountPoint, String alluxioRo
   public void umountFuse(String mountPath) throws Exception {
     mFuseFileSystem.umount(true);
   }
+
+  /**
+   * Tests creating a file for writing
+   * and opening a file for O_RDONLY read-only open flag.
+   */
+  @Test
+  public void createWriteOpenRead() throws Exception {
+    String testFile = "/createWriteOpenReadFile";
+    int fileLen = 128;
+    try (CloseableFuseFileInfo info = new CloseableFuseFileInfo()) {
+      FuseFileInfo fuseFileInfo = info.getFuseFileInfo();
+      fuseFileInfo.flags.set(OpenFlags.O_RDONLY.intValue());
+      // cannot open non-existing file for read
+      Assert.assertNotEquals(0, mFuseFileSystem.open(testFile, fuseFileInfo));
+      // open existing file for read
+      createTestFile(testFile, fuseFileInfo, fileLen);
+      readAndValidateTestFile(testFile, fuseFileInfo, fileLen);
+    }
+  }
+
+  /**
+   * Tests opening a file for O_WRONLY write-only open flag.
+   */
+  @Test
+  public void openWrite() throws Exception {
+    String testFile = "/openWriteFile";
+    int fileLen = 128;
+    try (CloseableFuseFileInfo closeableFuseFileInfo = new CloseableFuseFileInfo()) {
+      FuseFileInfo info = closeableFuseFileInfo.getFuseFileInfo();
+      info.flags.set(OpenFlags.O_WRONLY.intValue());
+      // O_WRONLY non-existing file will write to file
+      Assert.assertEquals(0, mFuseFileSystem.open(testFile, info));
```

Comment:
WorkerFuse doesn't create `mFuseFileSystem` upon mounting. There are test failures related to this

Reply:
Good catch, revert the inheritance, thanks